data/network/src/commonMain/kotlin/social/androiddev/common/network/util/RunCatching.kt

```diff
@@ -0,0 +1,46 @@
+package social.androiddev.common.network.util
+
+import kotlinx.coroutines.CancellationException
+import kotlinx.coroutines.TimeoutCancellationException
+
+inline fun <reified T> runCatchingRetry(maxRetry: Int, block: () -> T): Result<T> {
+    var result: Result<T>
+    var retryCount = 0
+    do {
+        result = runCatching(block)
+        retryCount++
```

Comment:
Something about these is a bit off-putting. Are these meant to run inside a coroutine?

Also, wondering if we want to enable a retry timeout via strategy pattern:

```
interface RetryTimeoutStrategy {
  fun getNextTimeout(lastTimeout: Long): Long
}

// Don't use this directly, you'll want to add a lower and upper bounds.
class ExponentialBackoffTimeoutStrategy {
  override fun getNextTimeout(lastTimeout: Long): Long = pow(lastTimeout, 2)
}
```

Reply:
I just added a bunch of common wrappers and inline functions that I use in different projects with `kotlin.Result`. I can remove these if we don't see value in them.

This isn't restricted to just coroutines, just to be able to inline a simple retry mechanism if needed with error catching

Reply:
Yeah it just kinda looks like something someone could accidentally call from the main thread

Reply:
Good observation @exallium  - I_can make this a `suspend` function to enforce that, and if the consumer really wanted to run on main thread they could use a wrapping Main dispatcher.

Reply:
Added some javadoc comments for clarity, we may end up deleting/removing these if not needed. But if we do end up wanting to leverage some type of retry mechanism, we can look at bringing in something like your suggestion or tweaking these 👍🏼